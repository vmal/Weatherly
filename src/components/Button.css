.Button{
    margin-top: 2%;
}