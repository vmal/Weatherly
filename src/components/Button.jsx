import React, { Component } from 'react';
import './Button.css';

class Button extends Component{
    render() {
        return(
            <div>
                <input className="Button" type="button" value={this.props.name}
                       onClick={this.props.onClick} />
            </div>
        );
    }
}

export default Button;