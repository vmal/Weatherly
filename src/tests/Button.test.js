import React from 'react';
import {shallow} from 'enzyme';
import Button from '../components/Button';

describe('test button component',()=>{
    it('should be defined', () => {
        expect(Button).toBeDefined();
    });

    it('should render correctly', () => {
        const tree = shallow(
            <Button name='Set Location' />
        );
        expect(tree).toMatchSnapshot();
    });

    it('should test the click event',()=>{
        const mockCallBack = jest.fn();
        const tree = shallow(<Button onClick={mockCallBack} name='Set Location'/>);
        tree.find('input').simulate('click');
        expect(mockCallBack).toHaveBeenCalled();
    });

    it('should have the button value',()=>{
        const tree = shallow(<Button name="Test"/>);
        expect(typeof(tree.find('input').props().value)).toBe('string');
        expect(tree.find('input').props().value).toBe('Test');
    });
});