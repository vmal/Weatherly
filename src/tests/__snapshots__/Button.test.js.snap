// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    name="Set Location"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        className="Button"
        type="button"
        value="Set Location"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "Button",
        "onClick": undefined,
        "type": "button",
        "value": "Set Location",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="Button"
          type="button"
          value="Set Location"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "Button",
          "onClick": undefined,
          "type": "button",
          "value": "Set Location",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
